clustered/client/src/main/java/org/ehcache/clustered/client/internal/store/ClusteredStore.java

```diff
@@ -152,44 +155,66 @@ public boolean containsKey(final K key) throws StoreAccessException {
   @Override
   public PutStatus put(final K key, final V value) throws StoreAccessException {
     putObserver.begin();
-    PutOperation<K, V> operation = new PutOperation<K, V>(key, value, timeSource.getTimeMillis());
-    ByteBuffer payload = codec.encode(operation);
-    Chain chain = storeProxy.getAndAppend(key.hashCode(), payload);
-    ResolvedChain<K, V> resolvedChain = resolver.resolve(chain, key, timeSource.getTimeMillis());
-    if(resolvedChain.getResolvedResult(key) == null) {
-      putObserver.end(StoreOperationOutcomes.PutOutcome.PUT);
-      return PutStatus.PUT;
-    } else {
-      putObserver.end(StoreOperationOutcomes.PutOutcome.REPLACED);
-      return PutStatus.UPDATE;
+    PutStatus status = silentPut(key, value);
+    switch (status) {
+      case PUT:
+        putObserver.end(StoreOperationOutcomes.PutOutcome.PUT);
+        break;
+      case UPDATE:
+        putObserver.end(StoreOperationOutcomes.PutOutcome.REPLACED);
+        break;
+      case NOOP:
+        putObserver.end(StoreOperationOutcomes.PutOutcome.NOOP);
+        break;
+      default:
+        throw new IllegalArgumentException("Invalid put status: " + status);
```

Comment:
I'm being picky here, but this should throw `AssertionError` instead of `IllegalArgumentException` as this exception can never be caused by user code, but only by an internal bug.